app/src/main/java/net/gsantner/markor/format/todotxt/TodoTxtTextActions.java

```diff
@@ -322,4 +334,151 @@ public boolean onLongClick(View v) {
+    }
+
+    /**
+     * This routine searches the current line for a due date date 'due:YYYY-mm-dd'.
+     * If found, the due date is parsed and presented to the user to be updated.
+     *
+     * If no due date is found, the user is presented with a dialog to enter a date, and
+     * 'due:YYYY-mm-dd' is appended to the line
+     *
+     * @param offset if not existing date found, today's date is offset by this amount
+     */
+    protected void updateOrAppendDueDate(final int offset) {
+
+        final int[] selection = StringUtils.getSelection(_hlEditor);
+        Editable text = _hlEditor.getText();
+
+        final int lineStart = StringUtils.getLineStart(text, selection[0]);
+        final int lineEnd = StringUtils.getLineEnd(text, selection[1]);
+        String line = text.subSequence(lineStart, lineEnd).toString();
+
+        Matcher match = SttCommander.PATTERN_DUE_DATE.matcher(line);
+        final boolean found = match.find();
+
+        Calendar calendar = Calendar.getInstance();
+        // Parse selection for date
+        try {
+            if (found) calendar.setTime(DATEF_YYYY_MM_DD.parse(match.group(0)));
+            else calendar.add(Calendar.DAY_OF_MONTH, offset);
+        } catch (ParseException e) {
+            // Regex failed?; should not be here
+            e.printStackTrace();
+        }
+
+        DatePickerDialog.OnDateSetListener listener = (view, year, month, day) -> {
+            Calendar fmtCal = Calendar.getInstance();
+            fmtCal.set(year, month, day);
+            String tagDate = String.format("due:%s", DATEF_YYYY_MM_DD.format(fmtCal.getTime()));
+            String newline;
+            if (found) {
+                newline = match.replaceFirst(tagDate);
+            } else {
+                //Append with space, if needed
+                newline = line + (line.endsWith(" ")? "" : " ") + tagDate;
+            }
+            text.replace(lineStart, lineEnd, newline);
+        };
+
+        DateFragment dateFragment = new DateFragment()
+                .setActivity(_activity)
+                .setListener(listener)
+                .setCalendar(calendar);
+
+        dateFragment.show(((FragmentActivity) _activity).getSupportFragmentManager(), "dateFragment");
+    }
+
+    /**
+     * A DialogFragment to manage showing a DatePicker
+     */
+    public static class DateFragment extends DialogFragment {
+
+        private DatePickerDialog.OnDateSetListener listener;
+        private Activity activity;
+        private int year;
+        private int month;
+        private int day;
+
+        public DateFragment() {
+            super();
+            setCalendar(Calendar.getInstance());
+        }
+
+        public DateFragment setListener(DatePickerDialog.OnDateSetListener listener) {
+            this.listener = listener;
+            return this;
+        }
+
+        public DateFragment setActivity(Activity activity) {
```

Comment:
isn't this info exposed via `getActivity()` already as long attached to a activity?